### PR TITLE
MGMT-8934:  Day 2 Workers - Conformance

### PR DESCRIPTION
**Pull request by paul-maidment**
This test is a periodic test against OCP 4.11
These tests are running against an AWS instance of a single node cluster. Similar to the openshift-e2e-aws-single-node workflow.
This deploys an instance of single node and then adds a worker node to the cluster by scaling the replicas of a worker machineset, running the conformance tests after adding the node. After this, two
more nodes are added followed by another round of conformance tests.

### CODE REVIEW

File: ci-operator/config/openshift/oc/openshift-oc-release-4.10.yaml [hunk 1]
```diff
@@ -23,6 +23,11 @@ binary_build_commands: make build --warn-undefined-variables
 build_root:
   from_repository: true
 images:
+- dockerfile_literal: |
```
Comment (osherdp) [inline ci-operator/config/openshift/oc/openshift-oc-release-4.10.yaml:26]: the great news are that this image has been created successfully. take a look at: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/25752/rehearse-25752-pull-ci-openshift-oc-release-4.10-images/1488587798242922496/artifacts/build-logs/cli-with-jq.log

the less good news (not that bad though) is that it did failed to access some sources that are configured (all those ``localdev-*-rpms``). So I'll suggest you continue with your work, but do drop a message on slack channel ``#forum-testplatform`` so that they can take a look and maybe remove those sources, or correct them

Comment (petr-muller) [reply ci-operator/config/openshift/oc/openshift-oc-release-4.10.yaml]: We don't own content of any images

File: ci-operator/config/openshift/release/openshift-release-master__nightly-4.10.yaml [hunk 2]
```diff
@@ -75,6 +79,17 @@ base_images:
...
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.12
+images:
+- dockerfile_literal: |
+    FROM cli
+    RUN dnf install -y jq
+  from: cli
+  to: cli-with-jq
```
[the remark this answers is not in the document]
Comment (paul-maidment) [reply ci-operator/config/openshift/release/openshift-release-master__nightly-4.10.yaml:92]: does this mean that I can't use dockerfile_literal to add `jq` to an image?
what is the recommended means by which to do this, do you have an example of where this has been done before?

[the remark this answers is not in the document]
Comment (paul-maidment) [reply ci-operator/config/openshift/release/openshift-release-master__nightly-4.10.yaml]: I'll fix this

File: ci-operator/step-registry/single-node/add-worker/2/single-node-add-worker-2-ref.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    namespace: ocp
+    name: cli-jq
+    tag: latest
+  commands: single-node-add-worker-2-commands.sh
+  resources:
+    requests:
+      cpu: 100m
+      memory: 100Mi
+  documentation: |-
+    The Single Node Add Worker step adds a 2nd day worker to the SNO cluster under test. Step 2
```
Comment (omertuc) [inline ci-operator/step-registry/single-node/add-worker/2/single-node-add-worker-2-ref.yaml:13]: Let's just write in the description that it's identical to 1 but has to be its own step due to CI limitations

File: ci-operator/step-registry/single-node/e2e-test2/test/single-node-e2e-test2-test-ref.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  timeout: 30800s
+  env:
+  - name: TEST_ARGS
+    default: ""
+    documentation: |-
+      Additional arguments to be passed to 'openshift-test'
+  - name: TEST_TYPE
+    default: "suite"
+    documentation: |
+      The type of test to perform.
```
Comment (omertuc) [inline ci-operator/step-registry/single-node/e2e-test2/test/single-node-e2e-test2-test-ref.yaml:15]: Same here, delete all these docs, just say that everything is the same as the original test step just duplicated due to CI limitations

File: ci-operator/step-registry/single-node/e2e-test2/test/single-node-e2e-test2-test-ref.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
+ref:
+  as: single-node-e2e-test2-test
```
Comment (omertuc) [inline ci-operator/step-registry/single-node/e2e-test2/test/single-node-e2e-test2-test-ref.yaml:2]: Let's call this e2e-test-2

File: ci-operator/config/openshift/release/openshift-release-master__nightly-4.11.yaml [hunk 6]
```diff
@@ -119,6 +119,13 @@ tests:
         VIP_DHCP_ALLOCATION=no
         PROXY=yes
     workflow: baremetalds-assisted
+- as: e2e-aws-single-node-worker
+  interval: 24h
+  steps:
+    cluster_profile: aws-2
+    env:
+      BASE_DOMAIN: aws-2.ci.openshift.org
+    workflow: openshift-e2e-aws-single-node-worker
```
Comment (omertuc) [inline ci-operator/config/openshift/release/openshift-release-master__nightly-4.11.yaml:128]: nit: should probably `...-single-node-workers` (plural)

**Comment by omertuc** [comment]
Please unhold when rehearsal is green (or has only uninteresting tests failing on SNO)

**Comment by vrutkovs** [comment]
/approve

Needs `build-clusters` fix, so you'd need LGTM again

**Comment by omertuc** [comment]
Results looking [good](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/25752/rehearse-25752-periodic-ci-openshift-release-master-nightly-4.11-e2e-aws-single-node-workers/1495818491259785216), unholding

**Comment by omertuc** [comment]
Sorry - forgot - we need a reporter_config entry for this job so we get Slack messages. @paul-maidment please add one, you can grep for `#single-node-ci` to see how it's done for other jobs

**Comment by omertuc** [comment]
Looks like your recent push deleted all reporter_configs:
image

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml [hunk 7]
```diff
@@ -19031,7 +19031,7 @@ periodics:
       - --confirm
```
Comment (omertuc) [inline ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:19031]: This file is full of seemingly unrelated diffs, can you sort it out?
